Realm/RLMAnalytics.mm

```diff
@@ -101,22 +105,99 @@ static auto RLMSysCtl(int *mib, u_int mibSize, size_t *bufferSize) {
 
 // Get the version of OS X we're running on (even in the simulator this gives
 // the OS X version and not the simulated iOS version)
-static NSString *RLMOSVersion() {
-    std::array<int, 2> mib = {{CTL_KERN, KERN_OSRELEASE}};
+static NSString *RLMHostOSVersion() {
+    size_t size;
+    sysctlbyname("kern.osproductversion", NULL, &size, NULL, 0);
+    char *model = (char*)malloc(size);
+    sysctlbyname("kern.osproductversion", model, &size, NULL, 0);
+    NSString *deviceModel = [NSString stringWithCString:model encoding:NSUTF8StringEncoding];
+    free(model);
+    return deviceModel;
+}
+
+static NSString *RLMHostArch() {
+    std::array<int, 2> mib = {{CTL_HW, HW_MACHINE}};
     size_t bufferSize;
     auto buffer = RLMSysCtl(&mib[0], mib.size(), &bufferSize);
     if (!buffer) {
         return nil;
     }
 
-    return [[NSString alloc] initWithBytesNoCopy:buffer.release()
-                                          length:bufferSize - 1
-                                        encoding:NSUTF8StringEncoding
-                                    freeWhenDone:YES];
+    NSString *n = [[NSString alloc] initWithBytesNoCopy:buffer.release()
+                                                 length:bufferSize - 1
+                                               encoding:NSUTF8StringEncoding
+                                           freeWhenDone:YES];
+    return n;
+}
+
+static NSString *RLMTargetArch() {
+    NSString *targetArchitecture;
+#if TARGET_CPU_X86
+    targetArchitecture = @"x86";
+#elif TARGET_CPU_X86_64
+    targetArchitecture = @"x86_64";
+#elif TARGET_CPU_ARM
+    targetArchitecture = @"arm";
+#elif TARGET_CPU_ARM64
+    targetArchitecture = @"arm64";
+#endif
+    return targetArchitecture;
+}
+
+static NSString *RLMXCodeVersion() {
```

Comment:
Are we mapping this clang version to Xcode version in our analytics dashboard?, Because if it is something hard to do there (the team doing doesn't have the resources), it will be better to do it in our side

Reply:
I removed it